src/prefect/task_engine.py

```diff
@@ -599,6 +634,11 @@ def handle_crash(self, exc: BaseException) -> None:
         self.set_state(state, force=True)
         self._raised = exc
 
+        self._span.record_exception(exc)
+        self._span.set_status(Status(StatusCode.ERROR, state.message))
+        self._span.end(time.time_ns())
+        self._span = None
```

Comment:
This looks like it's repeated four times. Could you encapsulate it as a method on the base class for readability?